src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -688,6 +692,39 @@ void network::add_default_output_chains() {
     }
 }
 
+void network::calculate_weights_cache_capacity() {
+    auto get_weights_size = [](const program_node& node) {
+        size_t weights_size = 0;
+        if (node.is_type<fully_connected>()) {
+            weights_size = node.as<fully_connected>().weights().get_output_layout().bytes_count();
+        } else if (node.is_type<convolution>()) {
+            weights_size = node.as<convolution>().weights().get_output_layout().bytes_count();
+        } else if (node.is_type<deconvolution>()) {
+            weights_size = node.as<deconvolution>().weights().get_output_layout().bytes_count();
+        }
+
+        return weights_size;
+    };
+
+    size_t total_weights_size = 0;
+    for (auto node : _program->get_processing_order())
+        total_weights_size += get_weights_size(*node);
+
+    total_weights_size *= _config.get_property(ov::streams::num);
```

Comment:
Don't we share the weights within streams? I think it could be done simply if we move the weight cache to node?

Reply:
@yeonbok  yes, it is a good idea, but it looks like we need to cover some other topics to implement this in the most effective way:
1) Pre-compiled reorders kernels sharing over streams (move in_memory_kernels_cache from network to program OR use existing program's implementations cache)
2) Provide safe cache's entry access (LRU's methods _get()_ (which may update existing cache entry and cause some problem for other streams by cached object replacemnt), _add()_) from multiple streams
3) Memory reuse related difficulties
And maybe something else...

Reply:
I agree to have a separate work for the improvement. BTW about memory reuse related difficulties, why there would be such a concern? The memories for weights (constant) will not be reused.